otherlibs/dthreads/event.ml

```diff
@@ -0,0 +1,276 @@
+(**************************************************************************)
```

Comment:
I just basically copied the `Event` library from **systhreads**. It's weird, actually, there are two `Event` library implementations already. This is a copy-paste that adds a third.